dataactvalidator/interfaces/validatorStagingInterface.py

```diff
@@ -1,77 +1,52 @@
 from dataactcore.models.stagingInterface import StagingInterface
-from dataactcore.models.stagingModels import FieldNameMap
-from dataactcore.utils.responseException import ResponseException
-from dataactcore.utils.statusCode import StatusCode
-from dataactvalidator.interfaces.validatorJobTrackerInterface import ValidatorJobTrackerInterface
-from sqlalchemy import MetaData, Table
-from sqlalchemy.exc import NoSuchTableError
+from dataactcore.models.stagingModels import Appropriation, ObjectClassProgramActivity, AwardFinancial, AwardFinancialAssistance
 
 
 class ValidatorStagingInterface(StagingInterface):
     """ Manages all interaction with the staging database """
 
-    def dropTable(self,table):
+    def insertSubmissionRecordByFileType(self, record, fileType):
+        """Insert a submitted file row into staging.
+
+        Args:
+            record: record to insert (Dict)
+            submissionId: submissionId associated with this record (int)
+            fileType: originating file type of the record (string)
         """
+        if fileType == "award":
+            rec = AwardFinancialAssistance(**record)
+        elif fileType == "award_financial":
+            rec = AwardFinancial(**record)
+        elif fileType == "appropriations":
+            rec = Appropriation(**record)
+        elif fileType == "program_activity":
+            rec = ObjectClassProgramActivity(**record)
+        self.session.add(rec)
+        self.session.commit()
```

Comment:
That is what I mean by batch inserting, in the code being replaced I believe we were doing an insert after every 100 lines.  I assume sqlalchemy is smart enough to put multiple adds into a single insert statement at commit time, but I'm not certain of this, we should check on that.  I believe we were doing it in raw sql before.  My concern about waiting until the end of the file is that breaks our current pattern of never trying to hold the entire file in memory.  If we do n lines at a time we can control how much memory we're using.  For the story for large files, DB-35 is a story for that that's been in there since pretty much the beginning of the project.